src/shogun/distance/Distance.cpp

```diff
@@ -216,6 +216,18 @@ float64_t CDistance::distance(int32_t idx_a, int32_t idx_b)
 	return compute(idx_a, idx_b);
 }
 
+void CDistance::run_distance_rhs(SGVector<float64_t>& result, const index_t idx_r_start, index_t idx_start, const index_t idx_stop, const int32_t idx_a)
+{
+	for(int32_t i=idx_r_start; idx_start < idx_stop; i++,idx_start++)
+        result.vector[i] = this->distance(idx_a,idx_start);
+}
+
+void CDistance::run_distance_lhs(SGVector<float64_t>& result, const index_t idx_r_start, index_t idx_start, const index_t idx_stop, const int32_t idx_b)
+{
+	for(int32_t i=idx_r_start; idx_start < idx_stop; i++,idx_start++)
```

Comment:
this should be `index_t` just as any indexing variable....
you can do `++i` and `++idx_start`